### PR TITLE
Assembly: Fix context menu 'Activate object' not working and causing crash

### PR DESCRIPTION
**Pull request by PaddleStroke**
Fix https://github.com/FreeCAD/FreeCAD/issues/16617

Fix https://github.com/FreeCAD/FreeCAD/issues/15706 also!

### CODE REVIEW

File: src/Mod/Assembly/Gui/ViewProviderAssembly.cpp [hunk 1]
```diff
@@ -112,6 +114,20 @@ QIcon ViewProviderAssembly::getIcon() const
...
+    auto func = new Gui::ActionFunction(menu);
+
+    QAction* act = menu->addAction(QObject::tr("Active object"));
+    act->setCheckable(true);
+    act->setChecked(isActivePart());
+    func->trigger(act, [this]() {
+        this->doubleClicked();
+    });
+
+    ViewProviderDragger::setupContextMenu(menu, receiver, member);
```
Comment (chennes) [inline src/Mod/Assembly/Gui/ViewProviderAssembly.cpp:128]: Did you intentionally skip over ViewProviderPart here? It makes the linter cranky.

Comment (PaddleStroke) [reply src/Mod/Assembly/Gui/ViewProviderAssembly.cpp:128]: Yes exactly it's intended. `ViewProviderPart ` is implementing a QAction that is setting the part as active. We are overriding this QAction here.

Comment (PaddleStroke) [reply src/Mod/Assembly/Gui/ViewProviderAssembly.cpp]: I have just added a `//NOLINT`

**Comment by chennes** [comment]
With nine commits I'd squash this on commit -- if you want to keep some of them, could you please clean up the history a bit? Otherwise let me know a squash is OK.

**Comment by PaddleStroke** [comment]
Squash is ok yes

**Comment by chennes** [comment]
Backported to 1.0

**Pull request merged**